src/sunmatrix/sparse/sunmatrix_sparse.c

```diff
@@ -55,17 +57,20 @@ static int format_convert(const SUNMatrix A, SUNMatrix B);
 SUNMatrix SUNSparseMatrix(sunindextype M, sunindextype N, sunindextype NNZ,
                           int sparsetype, SUNContext sunctx)
 {
+  SUNFunctionBegin(sunctx);
   SUNMatrix A;
   SUNMatrixContent_Sparse content;
 
   /* return with NULL matrix on illegal input */
-  if ((M <= 0) || (N <= 0) || (NNZ < 0)) { return (NULL); }
-  if ((sparsetype != CSC_MAT) && (sparsetype != CSR_MAT)) { return (NULL); }
+  SUNAssertNull(M > 0 && N > 0, SUN_ERR_ARG_OUTOFRANGE);
+  SUNAssertNull(NNZ >= 0, SUN_ERR_ARG_OUTOFRANGE);
+  SUNAssertNull(sparsetype == CSC_MAT || sparsetype == CSR_MAT,
+                SUN_ERR_ARG_OUTOFRANGE);
```

Comment:
Should be `SUN_ERR_ARG_INCOMPATIBLE` or `SUN_ERR_ARG_WRONGTYPE` instead?

Reply:
I think `OUTOFRANGE` probably makes the most sense because it is an `int` and the user might be passing an `int` directly instead of one of the macros.